custom_components/better_thermostat/models/models.py

```diff
@@ -33,29 +35,33 @@ def convert_inbound_states(self, state):
 		if config.get('mode_map') is not None and state.get('system_mode') is not None:
 			hvac_mode = mode_remap(hvac_mode, reverse_modes(config.get('mode_map')))
 	
-	return {"current_heating_setpoint": current_heating_setpoint, "local_temperature": state.get('local_temperature'), "local_temperature_calibration": state.get('local_temperature_calibration'),
-		"system_mode"                 : hvac_mode}
+	return {
+		"current_heating_setpoint"     : current_heating_setpoint,
+		"local_temperature"            : state.get('local_temperature'),
+		"local_temperature_calibration": state.get('local_temperature_calibration'),
+		"system_mode"                  : hvac_mode}
 
 
-def get_device_model(self):
+async def get_device_model(self):
 	"""Fetches the device model from HA."""
-	
 	if self.model is None:
 		try:
-			if self.hass.states.get(self.heater_entity_id).attributes.get('device') is not None:
-				self.model = self.hass.states.get(self.heater_entity_id).attributes.get('device').get('model')
-			else:
-				raise ValueError
+			entity_reg = await er.async_get_registry(self.hass)
+			entry = entity_reg.async_get(self.heater_entity_id)
+			dev_reg = await dr.async_get_registry(self.hass)
+			device = dev_reg.async_get(entry.device_id)
+			try:
+				return re.search('\((.+?)\)', device.model).group(1)
+			except AttributeError:
+				return device.model
```

Comment:
This might throw an error.

Why do we need two trys here? Just log the error and return None if anything goes wrong.

Reply:
Thats for devices without a short name

Reply:
Ah okay?

Are you sure you get the right field from HA? There's definitely a field which Justin contains the model and one for the vendor somewhere stored in HA (maybes that's not accessible).

I'll have a look at this this evening, maybe there's a better way :)

Reply:
Make a try e.g. _LOGGER.debug("device: %s", device) in line 52. i think its a Z2M thing